src/Bicep.Core/Semantics/FunctionOverloadBuilder.cs

```diff
@@ -62,14 +65,14 @@ public FunctionOverloadBuilder WithDescription(string description)
         public FunctionOverloadBuilder WithReturnType(TypeSymbol returnType)
         {
             this.ReturnType = returnType;
-            this.ReturnTypeBuilder = args => returnType;
+            this.ReturnTypeBuilder = (_, _, _, _, _) => returnType;
 
             return this;
         }
 
-        public FunctionOverloadBuilder WithDynamicReturnType(FunctionOverload.ReturnTypeBuilderDelegate returnTypeBuilder)
+        public FunctionOverloadBuilder WithDynamicReturnType(FunctionOverload.ReturnTypeBuilderDelegate returnTypeBuilder, TypeSymbol signatureType)
```

Comment:
>signatureType [](http://example.com/codeflow?start=134&length=13)

Why do we need this?

Reply:
to be used as described here: https://github.com/Azure/bicep/pull/2501#discussion_r647987070